compiler/qsc_passes/src/capabilitiesck.rs

```diff
@@ -168,6 +172,33 @@ pub enum Error {
     UseOfClosure(#[label] Span),
 }
 
+/// Lower a package store from `qsc_frontend` HIR store to a `qsc_fir` FIR store.
+pub fn lower_store(
+    package_store: &qsc_frontend::compile::PackageStore,
+) -> qsc_fir::fir::PackageStore {
+    let mut fir_store = qsc_fir::fir::PackageStore::new();
+    for (id, unit) in package_store {
+        let package = qsc_lowerer::Lowerer::new().lower_package(&unit.package);
+        fir_store.insert(map_hir_package_to_fir(id), package);
+    }
+    fir_store
+}
+
+pub fn run_rca_pass(
+    fir_store: &qsc_fir::fir::PackageStore,
+    package_id: qsc_fir::fir::PackageId,
+    capabilities: RuntimeCapabilityFlags,
+) -> Vec<crate::Error> {
```

Comment:
If this functoion returned a tuple of `(Option<PackageStoreComputeProperties>, Vec<crate::Error>)` or some wrapper of that type, then the callers could return that as well and choose to reuse it when calling into `hir_to_qir` or `fir_to_qir`, which would save having to generate it twice. Right now the code runs `analyze_all` once to check errors and if there are none it runs again to prepare for partial evaluation.